### DISCUSSION TITLE
'vue-demi' causes errors when importing into Vitest test (only the .mjs file breaks it)

### DISCUSSION THREAD

**Discussion by dmix** [vuejs/pinia#1525] [Help and Questions]
### Describe the bug

I get this error whenever I import a Pinia store into my test:

```shell
SyntaxError: The requested module 'vue-demi' does not provide an export named 'computed'
 ❯ ../../async /Users/dmix/dev/callpixels/app/javascript/__tests__/stores/target_options_test.ts:1:256

```

Also got a more detailed version with some tweaks:

```shell
SyntaxError: Named export 'computed' not found. The requested module 'vue-demi' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'vue-demi';
const { getCurrentInstance, inject, toRaw, watch, unref, markRaw, effectScope, ref, isVue2, isRef, isReactive, set, onUnmounted, reactive, toRef, del, nextTick, computed, toRefs } = pkg;

 ❯ ../../async /Users/dmix/dev/callpixels/app/javascript/__tests__/stores/target_options_test.ts:1:256
```

This is the code in the test file (removing pinia stuff resolves the error)

```javascript
import { setActivePinia, createPinia } from 'pinia'
import {
    useTargetOptionStore,
    TargetOptionState,
} from '../../stores/target_options'

describe('TargetOption Store', function() {
    beforeEach(() => {
        setActivePinia(createPinia())
        store = useTargetOptionStore()
    })
    ...

```

This is not a new issue, others have managed to solve it by explicitly adding .mjs settings to webpack or nuxt config:

https://github.com/vueuse/vueuse/issues/718#issuecomment-946508863

The problem is that I'm using Vite and not webpack, so I dont know how to apply a similar config setting.

I resolved the issue by simply deleting the .mjs file `rm ./node_modules/pinia/dist/pinia.mjs`, this is fine because there's also the exact same .js version of this file that DOES import properly

Is there a way to fix this `mjs` issue?

### Reproduction

My full vite.config.ts file:

https://gist.github.com/dmix/780c81495d3fdfdf140e7cbe1e95e4c0

### Expected behavior

The pinia.mjs file should be imported properly allowing the tests to run

### Actual behavior

Error message whenever running `yarn run vitest run`

**Comment by MatusMak** [comment]
If anybody is experiencing this, it is caused by https://github.com/vueuse/vue-demi package and has been fixed in the release https://github.com/vueuse/vue-demi/releases/tag/v0.11.0

For me, all I had to do was to install `vue-demi` directly as my package to force the latest version, which resolved the issue.
